src/metalperformanceshaders.cs

```diff
@@ -3756,5 +3915,634 @@ interface MPSMatrixFindTopK {
 		[return: Release ()]
 		MPSMatrixFindTopK Copy ([NullAllowed] NSZone zone, [NullAllowed] IMTLDevice device);
 	}
+
+	[TV (11,3), iOS (11,3), Mac (10,13,4, onlyOn64: true)]
+	[BaseType (typeof (NSObject))]
+	interface MPSStateResourceList 	{
+
+		[Static]
+		[Export ("resourceList")]
+		MPSStateResourceList Create ();
+
+		// The following two variadic selectors are manually bound in MPSStateResourceList.cs
+		//[Static, Internal]
+		//[Export ("resourceListWithTextureDescriptors:", IsVariadic = true)]
+		//MPSStateResourceList ResourceListWithTextureDescriptors (MTLTextureDescriptor descriptor, IntPtr varArgs);
+
+		//[Static, Internal]
+		//[Export ("resourceListWithBufferSizes:", IsVariadic = true)]
+		//MPSStateResourceList ResourceListWithBufferSizes (nuint firstSize, IntPtr varArgs);
+
+		[Export ("appendTexture:")]
+		void Append (MTLTextureDescriptor descriptor);
+
+		[Export ("appendBuffer:")]
+		void Append (nuint size);
```

Comment:
I think this would make it clearer what the API does:

```suggestion
		void Append (nuint bufferSize);
```